site/app/routes/index.jsx

```diff
@@ -7,22 +7,21 @@ import { SectionHero } from '../components/section-hero.jsx';
 import { SectionNextEpisode } from '../components/section-next-episode.jsx';
 import { loadAllEpisodes } from '../util/load-all-episodes.server.js';
 
-export const loader = async () => {
-  const episodes = await loadAllEpisodes();
-
-  const schedule = await fetch(
-    'https://www.learnwithjason.dev/api/schedule',
-  ).then((res) => res.json());
-
-  const featured = await fetch(
-    'https://www.learnwithjason.dev/api/episodes/featured',
-  ).then((res) => res.json());
+const fetchJson = (url) => fetch(url).then((res) => res.json());
 
-  const sponsors = await fetch(
-    `https://www.learnwithjason.dev/api/sponsors`,
-  ).then((res) => res.json());
-
-  return { episodes, schedule, featured, sponsors };
+export const loader = async () => {
+  // we don't `await` on these until below so they can all be fetched in parallel
+  const episodesPromise = loadAllEpisodes();
+  const schedulePromise = fetchJson('https://www.learnwithjason.dev/api/schedule');
+  const featuredPromise = fetchJson('https://www.learnwithjason.dev/api/episodes/featured');
+  const sponsorsPromise = fetchJson('https://www.learnwithjason.dev/api/sponsors');
+
+  return { 
+    episodes: await episodesPromise,
+    schedule: await schedulePromise, 
+    featured: await featuredPromise, 
+    sponsors: await sponsorsPromise,
+  };
```

Comment:
Hi! I'm another internet stranger :). This code does not actually parallelize (unless Remix is doing some compiling under the hood that I don't know about!) If you want to parallelize, use

```js
await Promise.all([
  episodesPromise,
  schedulePromise,
  // ...etc
])
```

Here's a demo I made that hopefully helps: https://jsfiddle.net/69f1h28d/1/

Reply:
actually, my code _does_ fetch things in parallel, I made another revision to your jsfiddle to show you
https://jsfiddle.net/rckx5Lms/2/

Reply:
the tl;dr is that
```JavaScript

  return { 
     a: (await makePromise(500, 'a')),
     b: (await makePromise(100, 'b')),
   }

// is different than

  const cPromise =  makePromise(5000, 'c')
  const dPromise =  makePromise(1000, 'd')
  return { 
     c: await cPromise,
     d: await dPromise
   };
}
```
The difference that that one `await`s on the same line as the promise is made, the other creates the 2 promises then `await`s them later. 

You're right that you can also use `Promise.all[` but that is more ceremony and less readable

anyway, feel free to take my suggestion. no worries if you don't. still a really cool site and I think remix is pretty neat!

Reply:
Huh, TIL! Thanks for the explanation 😃 

(PS, not my site I just popped in here on idle time (...and exercise [Cunningham's Law](https://meta.wikimedia.org/wiki/Cunningham%27s_Law) apparently!))

Reply:
oh dang! I, too, thought you had to use `Promise.all` for this

thanks so much, @ryankshaw!